app/models/game.rb

```diff
@@ -8,4 +8,30 @@ class Game < ApplicationRecord
           OR
           (white_player_id IS NOT NULL AND black_player_id IS NULL)')
         }
+
+  class Board
+    attr_reader :grid
+    
+    def initialize(pieces)
+      @pieces = pieces
+      @grid = Array.new(8) { Array.new(8) }
+      fill_grid
+    end
+    
+    private
+    
+    def pieces
+      @pieces
+    end
+    
+    def fill_grid
+      pieces.each do |piece|
+        next if piece.is_captured
```

Comment:
Yes, I had forgotten that we talked about this in hangouts. Should the this class rely on its instantiator to pass the scoped pieces? E.g. `Game::Board.new(my_pieces.is_captured)`